src/sessions.ts

```diff
@@ -286,10 +302,6 @@ export class ClientSession
         this.hasEnded = true;
         this.emit('ended', this);
       }
-    } catch (error) {
-      // spec indicates that we should ignore all errors for `endSessions`
-      squashError(error);
-    } finally {
       maybeClearPinnedConnection(this, { force: true, ...options });
```

Comment:
Unrelated to this PR but maybeClearPinnedConnection can conceivably throw - if it does, it will override any other errors that were being thrown.  Should we  try-catch this and ensure we always throw the correct error (which I think would probably be an aggregation of both errors)?

Reply:
Wouldn't this be better as a [SuppressedError](https://github.com/tc39/proposal-explicit-resource-management#the-suppressederror-error)? 

```
let abortTransactionError = null; // make a let var at the top to capture the value from the catch

    try {
        maybeClearPinnedConnection(this, { force: true, ...options });
      } catch (error) {
        // eslint-disable-next-line no-unsafe-finally
        throw abortTransactionError != null
          ? new MongoSuppressedError(
              error,
              abortTransactionError,
              'Clearing pinned connections unexpectedly failed'
            )
          : error;
      }
```

I gave it a bit of a whirl, it's kinda involved, I am interested in looking into this further, but it is kind of a general hazzard to anywhere we use `finally` and assume we don't throw. Maybe we can introduce `MongoSuppressedError` separately and think about how we want to organize the code (like just a `let` or some common pattern to make capturing the errors consistent)

Reply:
From slack: follow up about disposeStack